File: solutions/julia/allergies/1/allergies.jl
allergies_dict = Dict([(0,"eggs"), (1,"peanuts"), (2, "shellfish"), (3, "strawberries"),
                        (4, "tomatoes"), (5, "chocolate"), (6, "pollen"), (7, "cats")])

function allergic_to(score, allergen)
     allergen in allergy_list(score)
end

function allergy_list(score)
    list = Set()
    for k in 0:7
       if score >> k & 1 == 1 
            push!(list, allergies_dict[k])
       end
    end
    list
end

File: solutions/python/allergies/1/allergies.py
class Allergies:

    def __init__(self, score):
        self.allergies_dict = { 0:"eggs", 1:"peanuts", 2:"shellfish", 3:"strawberries",
4:"tomatoes", 5:"chocolate", 6:"pollen", 7: "cats"}
        self.score = score
        
    def allergic_to(self, item):
        return item in self.lst

    @property
    def lst(self):
        result = []
        for k in range(0,8):
            if self.score>>k&1==1:
                result.append(self.allergies_dict[k])
        return result
